[WPE] WebKitImage uses incorrect color type when encoding

spenap opened the pull request:
#### bd53bc0caa62ef3e0ff79cead06684acd77441f8
<pre>
[WPE] WebKitImage uses incorrect color type when encoding
<a href="https://bugs.webkit.org/show_bug.cgi?id=303230">https://bugs.webkit.org/show_bug.cgi?id=303230</a>

Reviewed by Adrian Perez de Castro and Carlos Garcia Campos.

Align WebKitImage with WebKit::WebImage, using BGRA8. This fixes PNG
encoding in WebKitImage load-related functions. Additionally rename
webkit_image_as_rgba_bytes into webkit_image_as_bytes, to stop implying
any pixel format.

Test: Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitImage.cpp
* Source/WebKit/UIProcess/API/glib/WebKitImage.cpp:
(webkit_image_as_bytes):
(webkitImageHash):
(webkitImageEqual):
(webkitImageLoad):
(webkit_image_as_rgba_bytes): Deleted.
* Source/WebKit/UIProcess/API/glib/WebKitImage.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitImage.cpp:
(testWebKitImagePropertiesConstruct):

Canonical link: <a href="https://commits.webkit.org/303675@main">https://commits.webkit.org/303675@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/76c071abad7fa40b32bc087952c1186509f48835

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/133071 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5572 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/44174 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140609 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/85108 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134941 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/6063 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/5435 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101766 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/69135 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/136018 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/4269 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/119249 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82564 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/4154 "Passed tests") | [❌ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/1741 "Found 1 new API test failure: TestWebKitAPI.ObscuredContentInsets.TopOverhangColorExtensionLayer (failure)") | [⏳ 🛠 wpe-cairo ](https://ews-build.webkit.org/#/builders/WPE-Cairo-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/113225 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/37365 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/143261 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/5242 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37944 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/110146 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/5324 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4498 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/110326 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28000 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4040 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115506 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/58856 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/5297 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33858 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/5138 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68749 "Built successfully") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/5386 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/5253 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitImage.cpp:
```diff
@@ -247,21 +247,20 @@ guint webkit_image_get_stride(WebKitImage* image)
  * with 8-bit premultiplied alpha, in the preferred byte order for
- * the architecture (typically ABGR8888 on little-endian hosts, and
- * RGBA8888 on big-endian ones).
+ * the architecture.
```
I would mention here that the pixel format is specifically BGRA8888 because, after all, that's the hardcoded format in `WebImage::create`.

spenap replied:
That's true for snapshots. Will it be true if for example we are using WebKitImage for a favicon image?

aperezdc replied:
Good point. Then the correct location for the comment about the format would be in the documentation string for `webkit_web_web_view_get_snapshot_finish()`, where the image is returned. Right?

spenap replied:
Yes, very good point. I'll add a note there :-)

aperezdc reviewed (approved):
LGTM, with a nit in the documentation.

spenap committed:
[WPE] WebKitImage uses incorrect color type when encoding
https://bugs.webkit.org/show_bug.cgi?id=303230

Reviewed by Adrian Perez de Castro and Carlos Garcia Campos.

Align WebKitImage with WebKit::WebImage, using BGRA8. This fixes PNG
encoding in WebKitImage load-related functions. Additionally rename
webkit_image_as_rgba_bytes into webkit_image_as_bytes, to stop implying
any pixel format.

Test: Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitImage.cpp
* Source/WebKit/UIProcess/API/glib/WebKitImage.cpp:
(webkit_image_as_bytes):
(webkitImageHash):
(webkitImageEqual):
(webkitImageLoad):
(webkit_image_as_rgba_bytes): Deleted.
* Source/WebKit/UIProcess/API/glib/WebKitImage.h.in:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitImage.cpp:
(testWebKitImagePropertiesConstruct):

Canonical link: https://commits.webkit.org/303675@main
